src/stores/HomeStore.js

```diff
@@ -358,35 +394,70 @@ class HomeStore {
+      this.transferEvents.affirmationCompleted.push({
+        blockNumber: event.blockNumber,
+        value: BN(fromDecimals(event.returnValues.value,this.tokenDecimals))
+      })
+    } else if (event.event === "HomeFeeUpdated") {
+      this.feeManager.homeHistoricFee.push({
+        blockNumber: event.blockNumber,
+        fee: new BN(fromWei(event.returnValues.fee, 'ether'))
+      })
+    } else if (event.event === "ForeignFeeUpdated") {
+      this.feeManager.foreignHistoricFee.push({
+        blockNumber: event.blockNumber,
+        fee: new BN(fromWei(event.returnValues.fee, 'ether'))
+      })
     }
   }
 
-  processLargeArrayAsync(array, fn, maxTimePerChunk) {
-    maxTimePerChunk = maxTimePerChunk || 16;
-    let index = 0;
+  calculateCollectedFees() {
+    if (!this.statistics.finished
+      || !this.rootStore.foreignStore.feeUpdatedEventsFinished
+      || !this.feeManager.feeManagerMode
+      || !this.rootStore.foreignStore.feeManager.feeManagerMode) {
+      setTimeout(() => { this.calculateCollectedFees() }, 1000)
+      return
+    }
+
+    const data = getRewardableData(this.feeManager, this.rootStore.foreignStore.feeManager)
+
+    this.depositFeeCollected.type = data.depositSymbol === 'home' ? this.symbol : this.rootStore.foreignStore.symbol
+    this.withdrawFeeCollected.type = data.withdrawSymbol === 'home' ? this.symbol : this.rootStore.foreignStore.symbol
+    this.depositFeeCollected.shouldDisplay = data.displayDeposit
+    this.withdrawFeeCollected.shouldDisplay = data.displayWithdraw
+
+    if (this.depositFeeCollected.shouldDisplay) {
+      const feeAppliedOnValue = !(this.rootStore.bridgeMode === BRIDGE_MODES.NATIVE_TO_ERC)
+        || !(this.feeManager.feeManagerMode === FEE_MANAGER_MODE.ONE_DIRECTION)
 
-    function now() {
-      return new Date().getTime();
+      processLargeArrayAsync(
+        this.transferEvents.userRequestForSignature,
+        calculateValueFee(data.homeHistoricFee, this.depositFeeCollected, feeAppliedOnValue),
+        () => {
+          this.depositFeeCollected.finished = true
+        })
+    } else {
+      this.depositFeeCollected.finished = true
     }
 
-    const doChunk = () => {
-      const startTime = now();
-      while (index < array.length && (now() - startTime) <= maxTimePerChunk) {
-        // callback called with args (value, index, array)
-        fn.call(null, array[index], index, array);
-        ++index;
-      }
-      if (index < array.length) {
-        setTimeout(doChunk, 0);
-      } else {
-        this.statistics.finished = true
-        this.statistics.totalBridged = this.statistics.depositsValue.plus(this.statistics.withdrawsValue)
-      }
+    if (this.withdrawFeeCollected.shouldDisplay) {
+      processLargeArrayAsync(
+        this.transferEvents.affirmationCompleted,
+        calculateValueFee(data.foreignHistoricFee, this.withdrawFeeCollected, false),
```

Comment:
Is this correct that the `feeApplied` parameter is `false` here but it is conditional for `userRequestForSignature`?

Reply:
Yes, that is correct. For `AffirmationCompleted` the value from event doesn't have the fee applied on any of the possible fee modes configurations.

Reply:
thanks